frontend/src/features/profile/manage/api/updateDrafts.ts

```diff
@@ -0,0 +1,245 @@
+    // Load draft
+    const { data: draft, error: loadError } = await supabase
+      .from('vendor_drafts')
+      .select('*')
+      .eq('id', draftId)
+      .single();
+
+    if (loadError || !draft) {
+      return { success: false, error: 'Draft not found' };
+    }
+
+    console.log('Pretend to publish draft to vendor:', draft);
+
+    // // Update vendor with draft data
+    // const { error: updateError } = await supabase
+    //   .from('vendors')
+    //   .update({
+    //     business_name: draft.business_name,
+    //     website: draft.website,
+    //     email: draft.email,
+    //     ig_handle: draft.ig_handle,
+    //     google_maps_place: draft.google_maps_place,
+    //     description: draft.description,
+    //     region: draft.region,
+    //     city: draft.city,
+    //     state: draft.state,
+    //     country: draft.country,
+    //     latitude: draft.latitude,
+    //     longitude: draft.longitude,
+    //     travels_world_wide: draft.travels_world_wide,
+    //     lists_prices: draft.lists_prices,
+    //     bridal_hair_price: draft.bridal_hair_price,
+    //     bridal_makeup_price: draft.bridal_makeup_price,
+    //     'bridal_hair_&_makeup_price': draft.bridal_hair_makeup_price,
+    //     bridesmaid_hair_price: draft.bridesmaid_hair_price,
+    //     bridesmaid_makeup_price: draft.bridesmaid_makeup_price,
+    //     'bridesmaid_hair_&_makeup_price': draft.bridesmaid_hair_makeup_price,
+    //     cover_image: draft.cover_image,
+    //     profile_image: draft.profile_image,
+    //     logo: draft.logo,
+    //     // Note: gis_computed, state_id, metro_id are computed by DB
+    //   })
+    //   .eq('id', draft.vendor_id);
+
+    // if (updateError) {
+    //   return { success: false, error: updateError.message };
+    // }
+
+    // Mark draft as published
+    await supabase
+      .from('vendor_drafts')
+      .update({ is_published: true })
+      .eq('id', draftId);
+
+    return { success: true };
+  } catch (error) {
+    console.error('Error publishing draft:', error);
+    return {
+      success: false,
+      error: error instanceof Error ? error.message : 'Unknown error'
+    };
+  }
+}
+
+/**
+ * Discard draft changes (revert to published version)
+ */
+export async function discardDraft(draftId: string): Promise<boolean> {
+  const supabase = await createClient();
+
+  try {
+    const { data: draft } = await supabase
+      .from('vendor_drafts')
+      .select('vendor_id')
+      .eq('id', draftId)
+      .single();
+
+    if (!draft) return false;
+
+    // Reload from vendor table
+    const { data: vendor } = await supabase
+      .from('vendors')
+      .select('*')
+      .eq('id', draft.vendor_id)
+      .single();
+
+    if (!vendor) return false;
+
+    // Reset draft to match vendor
+    await supabase
+      .from('vendor_drafts')
+      .update({
+        business_name: vendor.business_name,
+        website: vendor.website,
+        // ... all other fields
```

Comment:
is this a todo?

Reply:
updated to reuse the updateVendor code from admin